txcache/virtualSessionComputer.go

```diff
@@ -0,0 +1,79 @@
+package txcache
+
+import (
+	"math/big"
+)
+
+type virtualSessionComputer struct {
+	session                  SelectionSession
+	validator                *breadcrumbsValidator
+	virtualAccountsByAddress map[string]*virtualAccountRecord
+}
+
+func newVirtualSessionComputer(session SelectionSession) *virtualSessionComputer {
+	return &virtualSessionComputer{
+		session:                  session,
+		validator:                newBreadcrumbValidator(),
+		virtualAccountsByAddress: make(map[string]*virtualAccountRecord),
+	}
+}
+
+// createVirtualSelectionSession iterates over the chain of tracked blocks and for each sender checks that its breadcrumb is continuous
+// If the breadcrumb of an account is continuous, the virtual record of that account is created or updated
+func (computer *virtualSessionComputer) createVirtualSelectionSession(
+	chainOfTrackedBlocks []*trackedBlock,
+) (*virtualSelectionSession, error) {
+
```

Comment:
Superfluous newline. Function header can also be on a single line (opinion).

Formatting (nitpicking).